File: Genetibase.MathX/Genetibase.MathX.NuGenCCalc_x64/src/Genetibase.MathX.NugenCCalc.Adapters/Series.vb
Imports System.ComponentModel
Imports System.Drawing

Public Class Series

    Private _index As Integer
    Private _label As String = "newSeries"
    Private _color As Color = Drawing.Color.Red

    <Browsable(False)> _
    Public Property Index() As Integer
        Get
            Return _index
        End Get
        Set(ByVal value As Integer)
            If (_index <> value) Then
                _index = value
            End If
        End Set
    End Property

    <DefaultValue("newSeries")> _
    Public Property Label() As String
        Get
            Return _label
        End Get
        Set(ByVal value As String)
            If (_label <> value) Then
                _label = value
            End If
        End Set
    End Property


    <TypeConverter(GetType(ColorConverter))> _
    Public Property Color() As Color
        Get
            Return _color
        End Get
        Set(ByVal value As Color)
            If (_color <> value) Then
                _color = value
            End If
        End Set
    End Property

    Public Sub New(ByVal index As Integer, ByVal label As String, ByVal color As Color)
        _index = index
        _label = label
        _color = color
    End Sub

    Public Sub New(ByVal label As String, ByVal color As Color)
        _index = -1
        _label = label
        _color = color
    End Sub

    Public Sub New(ByVal label As String)
        _index = -1
        _label = label
        _color = Drawing.Color.Black
    End Sub

    Public Overrides Function ToString() As String
        Return _label
    End Function

End Class

File: Genetibase.MathX/Genetibase.MathX.NuGenCCalc_x86/src/Genetibase.MathX.NugenCCalc/Series.cs
//using System;
//using System.ComponentModel;
//using System.Drawing;

//namespace Genetibase.MathX.NugenCCalc
//{
//    /// <summary>
//    /// Store settings of series for 2D charts
//    /// </summary>
//    /// <example>
//    /// 	<code lang="CS" description="The following example set new series for NugenCCalcComponent">
//    /// NugenCCalcComponent ccalc = new NugenCCalcComponent();
//    /// ccalc.Series = new Series("newSeries", Color.Red);
//    /// </code>
//    /// </example>
//    [Serializable()]
//    public class Series
//    {
//        private int _index;
//        private string _label;
//        private Color _color;

//        /// <summary>
//        /// Gets or sets the index of the series in the chart
//        /// </summary>
//        [Browsable(false)]
//        public int Index
//        {
//            get
//            {
//                return _index;
//            }
//            set
//            {
//                if (_index != value)
//                {
//                    _index = value;
//                }
//            }
//        }

//        /// <summary>
//        /// Gets or sets the name of the series displayed by the chart
//        /// </summary>
//        [DefaultValue("newSeries")]
//        public string Label
//        {
//            get
//            {
//                return _label;
//            }
//            set
//            {
//                if (this._label != value)
//                {
//                    this._label = value;
//                }
//            }
//        }

//        /// <summary>
//        /// Gets or sets the color of the series displayed by the chart
//        /// </summary>
//        [TypeConverter(typeof(ColorConverter))]
//        public Color Color
//        {
//            get
//            {
//                return _color;
//            }
//            set
//            {
//                if (this._color != value)
//                {
//                    this._color = value;
//                }
//            }
//        }

//        /// <summary>
//        /// Initializes a new instance of the Series class on the specified index, label and color. 
//        /// </summary>
//        /// <param name="index">index of series</param>
//        /// <param name="label">label of series</param>
//        /// <param name="color">color of series</param>
//        public Series(int index, string label, Color color)
//        {
//            _index = index;
//            _label = label;
//            _color = color;
//        }

//        /// <summary>
//        /// Initializes a new instance of the Series class on the specified label and color. 
//        /// </summary>
//        /// <param name="label">label of series</param>
//        /// <param name="color">color of series</param>
//        public Series(string label, Color color)
//        {
//            _index = -1;
//            _label = label;
//            _color = color;
//        }

//        /// <summary>
//        /// Initializes a new instance of the Series class on the specified label.
//        /// </summary>
//        /// <param name="label">label of series</param>
//        public Series(string label)
//        {
//            _index = -1;
//            _label = label;
//            _color = Color.Black;
//        }


//        /// <summary>
//        /// Overridden. Returns label for current series
//        /// </summary>
//        public override string ToString()
//        {
//            return _label;
//        }

//    }
//}
